src/output-json.c

```diff
@@ -152,6 +152,86 @@ json_t *JsonAddStringN(const char *string, size_t size)
     return SCJsonString(tmpbuf);
 }
 
+void EveFileInfo(JsonBuilder *jb, const File *ff, const bool stored)
+{
+    size_t filename_size = ff->name_len * 2 + 1;
+    char filename_string[filename_size];
+    BytesToStringBuffer(ff->name, ff->name_len, filename_string, filename_size);
+    jb_set_string(jb, "filename", filename_string);
+
+    jb_open_array(jb, "sid");
+    for (uint32_t i = 0; ff->sid != NULL && i < ff->sid_cnt; i++) {
+        jb_append_uint(jb, ff->sid[i]);
+    }
+    jb_close(jb);
+
+#ifdef HAVE_MAGIC
+    if (ff->magic)
+        jb_set_string(jb, "magic", (char *)ff->magic);
+#endif
+    jb_set_bool(jb, "gaps", ff->flags & FILE_HAS_GAPS);
+    switch (ff->state) {
+        case FILE_STATE_CLOSED:
+            jb_set_string(jb, "state", "CLOSED");
+#ifdef HAVE_NSS
+            if (ff->flags & FILE_MD5) {
+                size_t x;
+                int i;
+                char str[256];
+                for (i = 0, x = 0; x < sizeof(ff->md5); x++) {
+                    i += snprintf(&str[i], 255-i, "%02x", ff->md5[x]);
+                }
+                jb_set_string(jb, "md5", str);
+            }
+            if (ff->flags & FILE_SHA1) {
+                size_t x;
+                int i;
+                char str[256];
+                for (i = 0, x = 0; x < sizeof(ff->sha1); x++) {
+                    i += snprintf(&str[i], 255-i, "%02x", ff->sha1[x]);
+                }
+                jb_set_string(jb, "sha1", str);
+            }
+#endif
+            break;
+        case FILE_STATE_TRUNCATED:
+            jb_set_string(jb, "state", "TRUNCATED");
+            break;
+        case FILE_STATE_ERROR:
+            jb_set_string(jb, "state", "ERROR");
+            break;
+        default:
+            jb_set_string(jb, "state", "UNKNOWN");
+            break;
+    }
+
+#ifdef HAVE_NSS
+    if (ff->flags & FILE_SHA256) {
+        size_t x;
+        int i;
+        char str[256];
+        for (i = 0, x = 0; x < sizeof(ff->sha256); x++) {
+            i += snprintf(&str[i], 255-i, "%02x", ff->sha256[x]);
+        }
+        jb_set_string(jb, "sha256", str);
+    }
+#endif
+
+    if (stored) {
+        jb_set_bool(jb, "stored", true);
+        jb_set_uint(jb, "file_id", ff->file_store_id);
+    } else {
+        jb_set_bool(jb, "stored", false);
```

Comment:
I guess we could now have a JB_SET_FALSE as well.

Reply:
I'll add this.